web/crux-ui/src/components/products/versions/images/config/crane-config-section.tsx

```diff
@@ -0,0 +1,277 @@
+import { EditorStateOptions } from '@app/components/editor/use-editor-state'
+import KeyOnlyInput from '@app/components/shared/key-only-input'
+import KeyValueInput from '@app/components/shared/key-value-input'
+import DyoChips from '@app/elements/dyo-chips'
+import { DyoHeading } from '@app/elements/dyo-heading'
+import { DyoInput } from '@app/elements/dyo-input'
+import { DyoLabel } from '@app/elements/dyo-label'
+import DyoSwitch from '@app/elements/dyo-switch'
+import { ImageConfigFilterType } from '@app/models'
+import {
+  ContainerConfig,
+  ContainerDeploymentStrategyType,
+  CONTAINER_DEPLOYMENT_STRATEGY_VALUES,
+  CraneConfigDetails,
+} from '@app/models/container'
+import { nullify } from '@app/utils'
+import useTranslation from 'next-translate/useTranslation'
+import { useState } from 'react'
+
+interface CraneConfigSectionProps {
+  config: CraneConfigDetails
+  onChange: (config: Partial<ContainerConfig>) => void
+  filters: ImageConfigFilterType[]
+  editorOptions: EditorStateOptions
+}
+
+const CraneConfigSection = (props: CraneConfigSectionProps) => {
+  const { t } = useTranslation('container')
+  const { config: propsConfig, filters, onChange: propsOnChange, editorOptions } = props
+
+  const [config, setConfig] = useState<CraneConfigDetails>(propsConfig)
+
+  const onChange = (newConfig: Partial<ContainerConfig>) => {
+    setConfig({ ...config, ...newConfig })
+    propsOnChange(newConfig)
+  }
+
+  const toNumber = (value: string): number => {
```

Comment:
This is also defined in CommonConfigSection, so I think it'd be better to move it to a utils file.